### ISSUE TITLE
Rename tab function not working

### ISSUE THREAD

**Issue by 117649** [onemen/TabMixPlus#39]
I'm trying to fix it. And got some positive result. Have you already worked on that?
Just want to be sure I'm not reinventing wheel.

I'll make a PR for the fix as soon as finish if it is needed.

**Comment by onemen** [comment]
You can work on it

**Comment by 117649** [comment]
#40
I got panel working.
But there is something in the callback hell of the set tab title would toggle the tittle between custom and original for every other reload of the page(f5, reload button) and load from url bar would loop it back to where it was(it still changes tittle you can see text is changing). Initial(restore session at browser startup) load would apply the user set tittle correctly so far . 

I just can't figure out how to solve it now.

The problem seems to be relate to:
https://github.com/onemen/TabMixPlus/blob/19bdd1b55b41ab935165cd7c3fc98e19147d0f55/addon/chrome/content/minit/tablib.js#L313

and:
https://github.com/onemen/TabMixPlus/blob/19bdd1b55b41ab935165cd7c3fc98e19147d0f55/addon/chrome/content/minit/tablib.js#L1629

**Comment by onemen** [comment]
I will look into it this weekend

**Comment by 117649** [comment]
I don't understand why this
https://github.com/onemen/TabMixPlus/blob/19bdd1b55b41ab935165cd7c3fc98e19147d0f55/addon/chrome/content/places/places.js#L384
is needed.
The tab seem to retain renamed title after restore from session.

**Comment by 117649** [comment]
> I will look into it this weekend

Anything?

**Comment by 117649** [comment]
https://github.com/onemen/TabMixPlus/blob/f5c1404f3c009c69ea83be5cff2afa3a1faeb52c/addon/chrome/content/minit/tablib.js#L1645-L1650
What things inside `.then()` is for?

I'm sure they were one factor of the problem and they were not in old code.

**Comment by 117649** [comment]
According to my understanding I propose change:
https://github.com/onemen/TabMixPlus/blob/f5c1404f3c009c69ea83be5cff2afa3a1faeb52c/addon/chrome/content/places/places.js#L410-L420
To:
```js
return this.asyncGetTabTitle(tab, url).then(newTitle => {
      // only call setTabTitle if we found one to avoid loop
      if (!newTitle) return false;
      if (newTitle != tab.label) {
        this.setTabTitle(tab, url, newTitle);
        if (initial) {
          tab._labelIsInitialTitle = true;
        }
      }
      return true;
    });
```
This prevent title from toggle while reload tab, However tab width still changes until loading is done then it fit to tittle width.

**Comment by onemen** [comment]
Fix by #40

Thank you @117649
